src/network-events.c

```diff
@@ -52,6 +52,47 @@ static __always_inline int save_sock_ptr(struct pt_regs *ctx, void *map)
     return 0;
 }
 
+typedef struct
+{
+    u16 protocol_type;           // Something like IPPROTO_TCP or IPPROTO_UDP
```

Comment:
Since we are changing the `remote_port` to be 0 for UDP, and "0" is not a port that a remote connection would have, is storing the `protocol_type` necessary at all? Might help us shave a bit of size out of the key.